src/main/java/seedu/duke/command/AddTaskCommand.java

```diff
@@ -21,9 +24,14 @@ public AddTaskCommand(String title, String dayOfTheWeek, String information) {
     }
 
     @Override
-    public void execute(Ui ui, TaskList tasklist, LessonList lessonList) {
+    public void execute(Ui ui, TaskList taskList, LessonList lessonList, Storage storage) throws DukeException {
         Task newTask = new Task(title, dayOfTheWeek, information);
-        tasklist.addTask(newTask);
-        ui.printTaskAdded(newTask, tasklist.getSize());
+        taskList.addTask(newTask);
+        ui.printTaskAdded(newTask, taskList.getSize());
+        try {
+            storage.saveData(taskList, lessonList);
+        } catch (IOException e) {
+            throw new DukeException(e.toString());
+        }
```

Comment:
`try-catch` might be unnecessary here. We can just throw both exceptions instead.

```java
public void execute(...) throws DukeException, IOException
```

Also, masking an `IOException` to a `DukeException` is probably not a good practice.